### ISSUE TITLE
Explicit mention of raised ValueError's after .detach() of underlying IO buffer

### ISSUE THREAD

**Issue by 3c565929-31be-498b-b929-dd4817d430a0** [python/cpython#81880]
BPO | [37699](https://bugs.python.org/issue37699)
--- | :---
Nosy | @bskinn, @iritkatriel

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2021-11-29.18:37:26.716>
created_at = <Date 2019-07-29.03:34:58.013>
labels = ['3.8', 'type-bug', '3.7', '3.9', 'docs']
title = "Explicit mention of raised ValueError's after .detach() of underlying IO buffer"
updated_at = <Date 2021-11-29.18:37:26.715>
user = 'https://github.com/bskinn'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-29.18:37:26.715>
actor = 'bskinn'
assignee = 'docs@python'
closed = True
closed_date = <Date 2021-11-29.18:37:26.716>
closer = 'bskinn'
components = ['Documentation']
creation = <Date 2019-07-29.03:34:58.013>
creator = 'bskinn'
dependencies = []
files = []
hgrepos = []
issue_num = 37699
keywords = []
message_count = 3.0
messages = ['348594', '407312', '407315']
nosy_count = 3.0
nosy_names = ['docs@python', 'bskinn', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = 'rejected'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue37699'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7', 'Python 3.8', 'Python 3.9']
```

</p></details>
